alluxio/__init__.py

```diff
@@ -0,0 +1 @@
+# This file is required for integration tests
```

Comment:
I'd like to target this for a later PR

Reply:
Without integration test it's will be hard to verify if this init action works. We need to tests at least happy-path that checks if cluster successfully created, similar to this one:
https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/blob/master/dr-elephant/test_dr_elephant.py

Reply:
added a minimal integration test. not sure it works though; it was looking for `alluxio.sh` init action in a bucket where it does not exist. how do i verify the test?

Reply:
Thanks for implementing test!

I have made few fixes to init action and refactored test so it could be executed with Bazel now:
```
bazel test --test_output=errors //alluxio:test_alluxio
```

But test is failing to create cluster:
```
+ /opt/alluxio/bin/alluxio formatWorker
Formatting Alluxio Worker @ test-alluxio-standard-1-3-20191025-184450-z7pi-w-1.us-central1-c.c.hadoop-cloud-dev.google.com.internal
2019-10-25 18:47:36,588 INFO  Format - Formatting worker data folder: /alluxioworker/
2019-10-25 18:47:36,589 INFO  Format - Formatting Data path for tier 0:/mnt/ramdisk/alluxioworker
2019-10-25 18:47:36,609 INFO  Format - Formatting complete
+ /opt/alluxio/bin/alluxio-start.sh worker NoMount
Killed 0 processes on test-alluxio-standard-1-3-20191025-184450-z7pi-w-1
Starting worker @ test-alluxio-standard-1-3-20191025-184450-z7pi-w-1.us-central1-c.c.hadoop-cloud-dev.google.com.internal. Logging to /opt/alluxio/logs
--- [ FAILED ] The worker @ test-alluxio-standard-1-3-20191025-184450-z7pi-w-1.us-central1-c.c.hadoop-cloud-dev.google.com.internal is not serving requests.
--- Printing the log tail for /opt/alluxio/logs/worker.log
>>> BEGIN
2019-10-25 18:47:39,911 WARN  RetryUtils - Failed to load cluster default configuration with master (attempt 1): alluxio.exception.status.UnavailableException: Failed to handshake with master test-alluxio-standard-1-3-20191025-184450-z7pi-m:19998 to load cluster default configuration values: UNAVAILABLE: io exception
```

Seems like failure is Alluxio-related, may you take a look what could be the reason for failure?

Reply:
Thanks for the fixes! I'll take a look at the error.

Reply:
@medb do you know how to resolve this error? i'm a newbie to bazel
```
12:41 $ bazel test --test_output=errors //alluxio:test_alluxio
Starting local Bazel server and connecting to it...
INFO: Writing tracer profile to '/private/var/tmp/_bazel_adit.madan/8f1f4e0e834e2f02ec623c993099deea/command.profile.gz'
INFO: Call stack for the definition of repository 'io_abseil_py' which is a http_archive (rule definition at /private/var/tmp/_bazel_adit.madan/8f1f4e0e834e2f02ec623c993099deea/external/bazel_tools/tools/build_defs/repo/http.bzl:292:16):
 - /Users/adit.madan/dataproc-fork/dataproc-initialization-actions/WORKSPACE:9:1
WARNING: Download from https://github.com/abseil/abseil-py/archive/master.zip failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException Checksum was a13c7b3fe98f90be3f6aa40044276e092ca25de6d98379884bdf4c4ecb79bfed but wanted 9b324bc96587819a1420c592cf5d54424456645719f1de6bb22f7045bb6dbc6b
ERROR: An error occurred during the fetch of repository 'io_abseil_py':
   java.io.IOException: Error downloading [https://github.com/abseil/abseil-py/archive/master.zip] to /private/var/tmp/_bazel_adit.madan/8f1f4e0e834e2f02ec623c993099deea/external/io_abseil_py/master.zip: Checksum was a13c7b3fe98f90be3f6aa40044276e092ca25de6d98379884bdf4c4ecb79bfed but wanted 9b324bc96587819a1420c592cf5d54424456645719f1de6bb22f7045bb6dbc6b
ERROR: /Users/adit.madan/dataproc-fork/dataproc-initialization-actions/alluxio/BUILD:3:1: //alluxio:test_alluxio depends on @io_abseil_py//absl/testing:parameterized in repository @io_abseil_py which failed to fetch. no such package '@io_abseil_py//absl/testing': java.io.IOException: Error downloading [https://github.com/abseil/abseil-py/archive/master.zip] to /private/var/tmp/_bazel_adit.madan/8f1f4e0e834e2f02ec623c993099deea/external/io_abseil_py/master.zip: Checksum was a13c7b3fe98f90be3f6aa40044276e092ca25de6d98379884bdf4c4ecb79bfed but wanted 9b324bc96587819a1420c592cf5d54424456645719f1de6bb22f7045bb6dbc6b
ERROR: Analysis of target '//alluxio:test_alluxio' failed; build aborted: no such package '@io_abseil_py//absl/testing': java.io.IOException: Error downloading [https://github.com/abseil/abseil-py/archive/master.zip] to /private/var/tmp/_bazel_adit.madan/8f1f4e0e834e2f02ec623c993099deea/external/io_abseil_py/master.zip: Checksum was a13c7b3fe98f90be3f6aa40044276e092ca25de6d98379884bdf4c4ecb79bfed but wanted 9b324bc96587819a1420c592cf5d54424456645719f1de6bb22f7045bb6dbc6b
INFO: Elapsed time: 5.065s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (14 packages loaded, 18 targets configured)
FAILED: Build did NOT complete successfully (14 packages loaded, 18 targets configured)
    Fetching @local_config_cc; fetching
```

Reply:
We have used Abseil dependency from the master HEAD. I have pinned Abseil version explicitly in #660, it should help.

Reply:
Thanks @medb. That did the trick. The test passes now:
```
22:21 $ bazel test --test_output=errors //alluxio:test_alluxio
INFO: Writing tracer profile to '/private/var/tmp/_bazel_adit.madan/8f1f4e0e834e2f02ec623c993099deea/command.profile.gz'
INFO: Analyzed target //alluxio:test_alluxio (0 packages loaded, 0 targets configured).
INFO: Found 1 test target...
Target //alluxio:test_alluxio up-to-date:
  bazel-bin/alluxio/test_alluxio
INFO: Elapsed time: 179.451s, Critical Path: 179.27s
INFO: 1 process: 1 local.
INFO: Build completed successfully, 2 total actions
//alluxio:test_alluxio                                                   PASSED in 179.3s

Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these arINFO: Build completed successfully, 2 total actions
```